pkg/guid/guid.go

```diff
@@ -0,0 +1,41 @@
+package guid
+
+import (
+	"fmt"
+	"net"
+)
+
+// GUID address is an uint64 encapsulation for network hardware address
+type GUID uint64
+
+const guidLength = 8
+
+// ParseGUID parses string only as GUID 64 bit
+func ParseGUID(s string) (GUID, error) {
+	ha, err := net.ParseMAC(s)
+	if err != nil {
+		return 0, err
+	}
+	if len(ha) != guidLength {
+		return 0, fmt.Errorf("invalid GUID address %s", s)
+	}
+
+	return GUID(uint64(ha[0])<<(7*guidLength) | uint64(ha[1])<<(6*guidLength) | uint64(ha[2])<<(5*guidLength) | uint64(ha[3])<<(4*guidLength) |
```

Comment:
can you do this in a for loop ?

Also i dont think you need to do this for big-endian CPU. do we support ppc64 with big endian architecture ?

Reply:
currently we don't support ppc64 so this is fine

Reply:
> can you do this in a for loop ?
I don't think it important for now or should block the patch

> 
> Also i dont think you need to do this for big-endian CPU. do we support ppc64 with big endian architecture ?

Reply:
> currently we don't support ppc64 so this is fine

that what blocked the patch :)

Reply:
yes :)